rpmlint/lint.py

```diff
@@ -81,6 +80,31 @@ def _load_installed_rpms(self, packages):
                 print_warning(f'(none): E: there is no installed rpm "{name}".')
         return existing_packages
 
+    def _load_rpmlintrc(self):
+        """
+        Load rpmlintrc from argument or load up from folder
+        """
+        if self.options['rpmlintrc']:
+            self.config.load_rpmlintrc(self.options['rpmlintrc'])
+        else:
+            # load only from the same folder specname.rpmlintrc or specname-rpmlintrc
+            # do this only in a case where there is one folder parameter or one file
+            # to avoid multiple folders handling
+            rpmlintrc = []
+            if not len(self.options['rpmfile']) == 1:
+                return
+            pkg = self.options['rpmfile'][0]
+            if pkg.is_file():
+                pkg = pkg.parent
+            rpmlintrc += sorted(pkg.glob('*.rpmlintrc'))
+            rpmlintrc += sorted(pkg.glob('*-rpmlintrc'))
```

Comment:
I would have liked a deprecation notice for any `rpmlintrc` files using `*-rpmlintrc`...

Reply:
scarabeus@bugaboo: ~/tmp/openSUSE:Factory $ grep '\-rpmlintrc' *.spec  |wc -l
793
scarabeus@bugaboo: ~/tmp/openSUSE:Factory $ grep '\.rpmlintrc' *.spec  |wc -l
67
The one with dash is winning by a huge margin.

Reply:
So? That doesn't make it right. It just means we have to support it while we tell everyone to fix their files.

Reply:
And what is the 'right' approach? Your feeling? SUSE is at the moment the only vendor doing this stuff and by majority it seems to be accepted that bla-rpmlintrc is the way how to denote them.
Yet you find .rpmlintrc more to your liking so we should force everyone to rename the files?

Reply:
> So? That doesn't make it right. It just means we have to support it while we tell everyone to fix their files.

it makes it right according to the openSUSE packaging guidelines

https://en.opensuse.org/openSUSE:Packaging_checks


we should simply ignore .rpmlintrc and consider them, according to the guidelines, as inexisting for the rpmlint checker)

Reply:
> And what is the 'right' approach? Your feeling? SUSE is at the moment the only vendor doing this stuff and by majority it seems to be accepted that bla-rpmlintrc is the way how to denote them.

Geez man, what the hell? Firstly, outside of the SUSE bubble, `<pkg>.rpmlintrc` is way more common. Secondly, in Fedora's infrastructure, you can use `.rpmlintrc` as a dot file for a package in Dist-Git. I know that this doesn't work in OBS VCS because its broken and can't handle dotfiles, but it's a thing. Thirdly, It's a bloody file extension to make it obvious that it's a file type "rpmlint runtime configuration".

It's not just a bloody feeling, it's based on the fact that it makes sense! Hell, before now, rpmlint didn't care at all what it was named, we just told it what to look up. You _changed_ that behavior without giving me a chance to respond until after the fact.

Reply:
I didn't change this approach, you can still denote it by your liking, but I added feautre to be able to process it based on the directory files if nothing is specified.
As we discussed in the ticket half a year agon we agreed to have -rpmlintrc and .rpmlintrc which will now work. It will parse even '.rpmlintrc' file.
So what actual problem your warning on -rpmlintrc will solve? Except you don't personaly like it?

Reply:
I didn't specifically agree to `<pkg>-rpmlintrc` in #125. But whatever.

I just wanted to compromise and give people time to migrate to a single predictable file name (`<srcpkgname>.rpmlintrc`) so that there'd be no chance of collisions. Allowing loading multiple `rpmlintrc` files can lead to weirdness, and the best way to eliminate that is to just have a single file we load per source package (+collection of binary packages).

Reply:
Actually loading multiple rpmlintrc files will not work ever, because what you want to shadow on one package will be needed on another.
Thats why I just make sure we found only one or warn otherwise, the same is that I don't look for them if multiple packages/folders are specified we don't look for them.

If you wan't to stricten the loader to be always `<srcpkg>.rpmlintrc` then be my guest but that should be another step where all the distros that are using the directory loading approach should update their policies.